### PR TITLE
Do not print sensitive properties' values in logs

### PR DESCRIPTION
**Pull request by kezhenxu94**
- [x] If this pull request closes/resolves/fixes an existing issue, replace the issue number. Closes #<issue number>.
- [x] Update the [`CHANGES` log](https://github.com/apache/skywalking/blob/master/docs/en/changes/changes.md).

### CODE REVIEW

File: oap-server/server-starter/src/main/java/org/apache/skywalking/oap/server/starter/config/ApplicationConfigLoader.java [hunk 1]
```diff
@@ -40,6 +43,15 @@
  */
 @Slf4j
 public class ApplicationConfigLoader implements ConfigLoader<ApplicationConfiguration> {
+    static final Set<String> SENSITIVE_PROPERTY_NAMES =
+        ImmutableSet.<String>builder()
+                    .add("password")
+                    .add("authentication")
+                    .add("accessKey")
+                    .add("secretKey")
```
Comment (toffentoffen) [inline oap-server/server-starter/src/main/java/org/apache/skywalking/oap/server/starter/config/ApplicationConfigLoader.java:51]: NIT: Check against https://skywalking.apache.org/docs/main/next/en/setup/backend/configuration-vocabulary/ and I think 1 is missing.
```suggestion
                    .add("secretKey")
                    .add("firehoseAccessKey")
```

**Review by toffentoffen — APPROVED** [review]
LGTM. Just a small suggestion to add 1 more sensitive property.

File: oap-server/server-starter/src/main/java/org/apache/skywalking/oap/server/starter/config/ApplicationConfigLoader.java (hunk 1 above)
Comment (kezhenxu94) [reply oap-server/server-starter/src/main/java/org/apache/skywalking/oap/server/starter/config/ApplicationConfigLoader.java:51]: @toffentoffen thank you!!

**Review by wankai123 — DISMISSED** [review]

**Comment by wu-sheng** [comment]
I can get the idea to.prevent data link, but the implementation seems not elegant.

Coukd you add an annotatio(SecurityConfig) for the module config field? And leverage that flag to output ******(if set) or empty

**Review by wu-sheng — CHANGES_REQUESTED** [review]
Hard code would not be a good practice, as OAP is high modularization and easy to add new module as extension.

**Comment by kezhenxu94** [comment]
> Hard code would not be a good practice, as OAP is high modularization and easy to add new module as extension.

What do you suggest?

**Comment by wu-sheng** [comment]
> I can get the idea to.prevent data link, but the implementation seems not elegant.
> 
> Coukd you add an annotatio(SecurityConfig) for the module config field? And leverage that flag to output ******(if set) or empty

I mean this.
Every module config should take their own responsibilities to declare this.
And the core only hide when needed.

**Comment by kezhenxu94** [comment]
We'd either not to log the configurations at all if you insist hardcode is not an option

**Comment by kezhenxu94** [comment]
> I can get the idea to prevent data leak, but the implementation seems not elegant.
> 
> Coukd you add an annotatio(SecurityConfig) for the module config field? And leverage that flag to output ******(if set) or empty

That's impossible, when printing the logs the configurations are still in `Map` type, there is no `ModuleConfig` type involved at this time, not to say a new annotation

**Comment by kezhenxu94** [comment]
At the same time I think it's unnecessary to do the complex work just for printing logs, I just removed the logs to print configurations.

**Comment by kezhenxu94** [comment]
> As a quick fix, the current implementation will work. However, I am totally with @wu-sheng suggestion to "modularize" it and wanted to suggest more or less the same. But as a follow up that could be tackled in the near future.
> 
> So, the final solution is to not print any configuration properties, until we implement in a standarized way, isn't it?

The logs are printed too early as soon as they are read, before they are deserialized into the configuration beans, so there is no way to add an annotation to those module config and leverage that annotation to determine whether or not to print the logs, if the logs are really needed they must be moved to somewhere after the configs are deserialized into the configuration beans

**Pull request merged**